web/src/app/add-category/add-category.component.ts

```diff
@@ -53,19 +54,20 @@ export class AddCategoryComponent implements OnInit {
   }
 
   add(event: MatChipInputEvent): void {
-    console.log("add");
-    let name = event.value.trim();
-    if (!name) {
-      return;
-    }
-    let category = this.findExistingCategory(name);
-    if (!category) {
-      this.persistCategory(name)
-    } else {
-      this.addCategory(category);
-    }
+    if (!this.matAutocomplete.isOpen) {
+      console.log("add");
+      let name = event.value.trim();
 
-    this.resetForm(event.input);
+      if (name || '') {
```

Comment:
Why do we check it? Does the form send submit when it empty?

Reply:
Ok, I'll examine this further.

Reply:
We can reset this change. Everything works the same.